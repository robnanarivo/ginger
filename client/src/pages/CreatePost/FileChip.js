import { React } from 'react';
import { Box, Chip } from '@mui/material';

const FileChip = ({ file, onDelete }) => (
  <Box
    sx={{
      display: 'flex',
      justifyContent: 'left',
      flexWrap: 'wrap',
      listStyle: 'none',
      p: 0.5,
      m: 0,
    }}
    component="ul"
  >
    <Chip
      label={file.file.name}
      variant="outlined"
      color="primary"
      onDelete={() => onDelete()}
    />
  </Box>
);

export default FileChip;
