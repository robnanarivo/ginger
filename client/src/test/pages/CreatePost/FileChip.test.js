import React from 'react';
import renderer from 'react-test-renderer';
import FileChip from '../../../pages/CreatePost/FileChip';
import 'jest-styled-components';

describe('FileChip unit test', () => {
  it('FileChip Snapshot test', () => {
    const file = {
      file: {
        name: 'good file',
      },
    };
    const tree = renderer
      .create(<FileChip file={file} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
