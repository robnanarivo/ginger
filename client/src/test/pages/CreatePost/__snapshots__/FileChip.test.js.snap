// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileChip unit test FileChip Snapshot test 1`] = `
<ul
  className="MuiBox-root css-a9m3sx"
>
  <div
    className="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-deletable MuiChip-deletableColorPrimary MuiChip-outlinedPrimary css-1l4dgo2-MuiButtonBase-root-MuiChip-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      good file
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorPrimary MuiChip-deleteIconOutlinedColorPrimary css-i4bv87-MuiSvgIcon-root"
      data-testid="CancelIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</ul>
`;
